src/sentry/workflow_engine/processors/detector.py

```diff
@@ -97,6 +114,107 @@ def _ensure_detector(project: Project, type: str) -> Detector:
         raise UnableToAcquireLockApiError
 
 
+def _ensure_metric_detector(
+    project: Project, owner_team_id: int | None = None, enabled: bool = True
+) -> Detector:
+    """
+    Ensure that a default anomaly detection metric monitor exists for a project.
+    If the Detector doesn't already exist, we try to acquire a lock to avoid double-creating.
+    """
+    # If it already exists, return immediately. Prefer the oldest if duplicates exist.
+    existing = (
+        Detector.objects.filter(type=MetricIssue.slug, project=project).order_by("id").first()
+    )
+    if existing:
+        return existing
+
+    lock = locks.get(
+        f"workflow-engine-project-{MetricIssue.slug}-detector:{project.id}",
+        duration=2,
+        name=f"workflow_engine_default_{MetricIssue.slug}_detector",
+    )
+    try:
+        with (
+            lock.blocking_acquire(initial_delay=0.1, timeout=3),
+            transaction.atomic(router.db_for_write(Detector)),
+        ):
+            # Double-check after acquiring lock in case another process created it
+            existing = (
+                Detector.objects.filter(type=MetricIssue.slug, project=project)
+                .order_by("id")
+                .first()
+            )
+            if existing:
+                return existing
+
+            condition_group = DataConditionGroup.objects.create(
+                logic_type=DataConditionGroup.Type.ANY,
+                organization_id=project.organization_id,
+            )
+
+            DataCondition.objects.create(
+                comparison={
+                    "sensitivity": AnomalyDetectionSensitivity.LOW,
+                    "seasonality": AnomalyDetectionSeasonality.AUTO,
+                    "threshold_type": AnomalyDetectionThresholdType.ABOVE,
+                },
+                condition_result=DetectorPriorityLevel.HIGH,
+                type=Condition.ANOMALY_DETECTION,
+                condition_group=condition_group,
+            )
+
+            detector = Detector.objects.create(
+                project=project,
+                name="High Error Count (Default)",
+                description="Automatically monitors for anomalous spikes in error count",
+                workflow_condition_group=condition_group,
+                type=MetricIssue.slug,
+                config={
+                    "detection_type": AlertRuleDetectionType.DYNAMIC.value,
+                    "comparison_delta": None,
+                },
+                owner_team_id=owner_team_id,
+                enabled=enabled,
+            )
+
+            snuba_query = create_snuba_query(
+                query_type=SnubaQuery.Type.ERROR,
+                dataset=Dataset.Events,
+                query="",  # filter????
```

Comment:
I think no query makes sense for a default rule. `is:unresolved` doesn't work for anomaly detection.